From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Visa Hankala <visa@hankala.org>
Date: Fri, 10 Jun 2022 13:25:13 +0000
Subject: [PATCH] pesigcheck: Fix crash on digest match

Set selected_digest when the digest is found in db or dbx.
This fixes the following crash of pesigcheck:

  Program received signal SIGSEGV, Segmentation fault.
  0x00005555555597fa in memcpy (__len=24, __src=0x31,
      __dest=0x55555558d908)
      at /usr/include/x86_64-linux-gnu/bits/string_fortified.h:34
  34      return __builtin___memcpy_chk (__dest, __src, __len, __bos0 (__dest));
  (gdb) bt
  #0  0x00005555555597fa in memcpy (__len=24, __src=0x31,
      __dest=0x55555558d908)
      at /usr/include/x86_64-linux-gnu/bits/string_fortified.h:34
  #1  get_digest (digest=digest@entry=0x55555558d908,
      ctx=<optimized out>, ctx=<optimized out>) at pesigcheck.c:226
  #2  0x00005555555592fd in check_signature (
      reasons=<synthetic pointer>, nreasons=<synthetic pointer>,
      ctx=0x7fffffffded0) at pesigcheck.c:262
  #3  main (argc=<optimized out>, argv=<optimized out>)
      at pesigcheck.c:512

Signed-off-by: Visa Hankala <visa@hankala.org>
---
 src/certdb.c | 8 ++++++--
 1 file changed, 6 insertions(+), 2 deletions(-)

NOTE: This change is already present in pesign 116. The patch body is a no-op so the packaging series continues to apply cleanly after the version bump.

diff --git a/TODO b/TODO
--- a/TODO
+++ b/TODO
@@ -1 +1 @@
-Stuff that needs to get done:
+Stuff that needs to get done:
